getting-started/kubernetes/vpp/baremetal.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Install VPP on a baremetal cluster
+description: Install Calico with the VPP dataplane on a baremetal cluster
+canonical_url: '/getting-started/kubernetes/vpp/baremetal'
+---
+
+### Requirements
+
+The following requirements are optional, but depending on your kernel version and hardware they may enable more efficient drivers to be used. To this end, ensure the following on all nodes:
+- At least 128 x 2MB-hugepages are available (`cat /proc/meminfo | grep HugePages_Free`)
+- The `vfio-pci` (named `vfio_pci` on centos) or `uio_pci_generic` kernel module is loaded
+- Swap is disabled (Kubernetes requirement which should already be true)
+
+This can be done with:
+````bash
+modprobe vfio-pci
+swapoff -a
+sysctl -w vm.nr_hugepages=128
+# restart kubelet to take the changes into account, you may need to use a different command depending on how kubelet was installed
+systemctl restart kubelet
+````
+
+You might also run with less requirements depending on your [primary interface configuration]({{ site.baseurl }}/reference/vpp/uplink-configuration)
+
+### Configuration
+
+Then start by getting the Calico-VPP YAML file:
+```bash
+# If you have hugepages available on your machines
+wget https://raw.githubusercontent.com/projectcalico/vpp-dataplane/release/\
+v0.12.1-calicov3.17.1/yaml/generated/calico-vpp.yaml
+# If not
+wget https://raw.githubusercontent.com/projectcalico/vpp-dataplane/release/\
+v0.12.1-calicov3.17.1/yaml/generated/calico-vpp-nohuge.yaml
+```
+
+Two things need to be configured in the ConfigMap that is defined at the bottom of this file:
+````yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: calico-config
+  namespace: kube-system
+data:
+  service_prefix: 10.96.0.0/12
+  vpp_dataplane_interface: eth1
+  default_ipv4_pool_cidr: 172.16.0.0/16
+  vpp_uplink_driver: ""
+````
+
+- `vpp_dataplane_interface` is the physical interface that VPP will use. This needs to be the name of a configured Linux interface. VPP will replicate the configuration of this interface, including addresses and routes.
+- `service_prefix` is the Kubernetes service CIDR. You can retrieve it by running:
+````sh
+kubectl cluster-info dump | grep -m 1 service-cluster-ip-range
+````
+- `default_ipv4_pool_cidr` allows you to customize the default Calico IPv4 pool. For more info check the [reference]({{ site.baseurl }}/reference/node/configuration)
+- `vpp_uplink_driver` Configures how VPP grabs the physical interface, available values are :
+  * `""` : will select try the drivers starting with the fastest
+  * `avf` : use the native AVF driver
+  * `virtio` : use the native virtio driver (requires hugepages)
+  * `af_xdp` : use AF_XDP sock family (require at least kernel 5.4)
+  * `af_packet` : use AF_PACKET sock family (slow but failsafe)
+  * `none` : dont configure connectivity
+
+### Deployment
+
+You can directly edit the yaml file and apply the configuration.
```

Comment:
To make it easy to apply changes to future versions of Calico-VPP,  we recommend creating a 
{% include open-new-window.html text='kustomization' url='https://github.com/projectcalico/vpp-dataplane/tree/master/yaml/overlays' %} with the specific parameters for your cluster. A kustomization creates the {{site.prodname}} resources and deploys a DaemonSet that runs VPP on all the nodes in the cluster, and configures the CNI to connect the new containers through VPP.

To apply the configuration using kustomize:

```
kubectl kustomize . | kubectl apply -f -
```
To apply the configuration using a basic .yaml file:
```
kubectl apply -f calico-base-latest.yaml
```